api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -240,6 +241,14 @@ func (mw *azureManagedControlPlaneWebhook) ValidateUpdate(ctx context.Context, o
 		}
 	}
 
+	if ptr.Deref(m.Spec.DNSPrefix, "") != ptr.Deref(old.Spec.DNSPrefix, "") {
```

Comment:
I think we want to check if either value is nil first as nil is different from an empty string. Otherwise, the user could set the value from nil to an empty string and pass the check.

Reply:
Even if the user sets an empty string, that is invalid as it will not pass the regex so he will not be able to pass the check as m.Validate calls m.ValidateDNSPrefix which is going to throw an error.

Reply:
Updated the test cases PTAL.

Reply:
Maybe we can leave a comment explaining what you described? In general, having this line rely on validation elsewhere could cause problems for maintainability since someone changing it in the future could forget.

Reply:
Is there a reason we can't leverage the existing `webhookutils.ValidateImmutable` helper for this?